### ISSUE TITLE
ACF fields on Menus and Menuitems

### ISSUE THREAD

**Issue by sarahannnicholson** [airesvsg/acf-to-rest-api#293]
So far this plugin has been terrific when trying to get ACF's on my pages / posts. 

I hoping to start a discussion about getting ACF's for menus or (specifically) menuitems. Wordpress supports creators to add ACF's to menus and menu items. 

It would make it much easier to add things like icons to the menuItems.

**Comment by airesvsg** [comment]
Hi @sarahannnicholson,

You need register the menu post type in rest api, follow the code below:

```PHP
add_action( 'init', function() {
	global $wp_post_types;

	$post_type = 'nav_menu_item';
	if ( isset( $wp_post_types[$post_type] ) ) {
		$wp_post_types[$post_type]->show_in_rest = true;
		$wp_post_types[$post_type]->rest_base = 'menus';
	}
} );
```

**Endpoints**
http://site.com/wp-json/wp/v2/menus
http://site.com/wp-json/acf/v3/menus

Thanks

**Comment by sarahannnicholson** [comment]
This is awesome! Thank you so much!

Was there some documentation somewhere regarding this that I missed?

**Comment by airesvsg** [comment]
There is nothing in the documentation about it. :disappointed:

**Comment by sarahannnicholson** [comment]
Hey @airesvsg 

Thanks so much for the code snippet! I was trying to get this working, and I came across a bit of a road block. 

I want to expose the ACF fields I added to my menu items. But `$wp_post_types[$post_type]` doesn't seem to have any ACF fields, so `$wp_post_types[$post_type]->show_in_rest = true;` this line of code isn't exposing anything new to the rest API. 

After adding the code to my plugin, I don't see any difference in the REST API when I go to `{my-wordpress-domain}/wp-json/wp-api-menus/v2/menus`. The code is being run though, cause I can `echo` variables and can see those changes. 

Also, I barely know php or how Wordpress works. I'm a complete noob.
